Core/include/Acts/TrackFinding/TrackSelector.hpp

```diff
@@ -0,0 +1,97 @@
+  struct Config {
+    // Minimum/maximum local positions.
+    double loc0Min = -std::numeric_limits<double>::infinity();
+    double loc0Max = std::numeric_limits<double>::infinity();
+    double loc1Min = -std::numeric_limits<double>::infinity();
+    double loc1Max = std::numeric_limits<double>::infinity();
+    // Minimum/maximum track time.
+    double timeMin = -std::numeric_limits<double>::infinity();
+    double timeMax = std::numeric_limits<double>::infinity();
+    // Direction cuts.
+    double phiMin = -std::numeric_limits<double>::infinity();
+    double phiMax = std::numeric_limits<double>::infinity();
+    double etaMin = -std::numeric_limits<double>::infinity();
+    double etaMax = std::numeric_limits<double>::infinity();
+    double absEtaMin = 0.0;
+    double absEtaMax = std::numeric_limits<double>::infinity();
+    // Momentum cuts.
+    double ptMin = 0.0;
+    double ptMax = std::numeric_limits<double>::infinity();
+
+    std::size_t minMeasurements = 0;
+  };
+
+  /// Constructor from a config object
+  /// @param config is the configuration object
+  TrackSelector(const Config& config) : m_cfg(config) {}
+
+  /// Select tracks from an input container and copy them into an output
+  /// container
+  /// @tparam input_tracks_t is the type of the input track container
+  /// @tparam output_tracks_t is the type of the output track container
+  /// @param inputTracks is the input track container
+  /// @param outputTracks is the output track container
+  template <typename input_tracks_t, typename output_tracks_t>
+  void selectTracks(const input_tracks_t& inputTracks,
+                    output_tracks_t& outputTracks) const;
+
+  /// Get readonly access to the config parameters
+  /// @return the config object
+  const Config& config() const { return m_cfg; }
+
+ private:
+  Config m_cfg;
+};
+
+template <typename input_tracks_t, typename output_tracks_t>
+void TrackSelector::selectTracks(const input_tracks_t& inputTracks,
+                                 output_tracks_t& outputTracks) const {
+  // helper functions to select tracks
+  auto checkMin = [](auto x, auto min) { return min <= x; };
+  auto within = [](double x, double min, double max) {
+    return (min <= x) and (x < max);
+  };
+  auto isValidTrack = [&](const auto& trk) {
+    const auto theta = trk.theta();
+    const auto eta = -std::log(std::tan(theta / 2));
+    return within(trk.transverseMomentum(), m_cfg.ptMin, m_cfg.ptMax) and
+           within(std::abs(eta), m_cfg.absEtaMin, m_cfg.absEtaMax) and
+           within(eta, m_cfg.etaMin, m_cfg.etaMax) and
+           within(trk.phi(), m_cfg.phiMin, m_cfg.phiMax) and
+           within(trk.loc0(), m_cfg.loc0Min, m_cfg.loc0Max) and
+           within(trk.loc1(), m_cfg.loc1Min, m_cfg.loc1Max) and
+           within(trk.time(), m_cfg.timeMin, m_cfg.timeMax) and
+           checkMin(trk.nMeasurements(), m_cfg.minMeasurements);
+  };
+
+  for (auto track : inputTracks) {
+    if (!isValidTrack(track)) {
```

Comment:
should we also make this one available too?

so having another member function `isValidTrack`

Reply:
Sure, why not.

Reply:
Done.